README.adoc

```diff
@@ -0,0 +1,247 @@
+// Create the executor that Reactor will subscribe on.
+BoundedScheduledThreadPoolExecutor executorService =
+        new BoundedScheduledThreadPoolExecutor(
+                100, new ScheduledThreadPoolExecutor(8));
+Scheduler scheduler = Schedulers.fromExecutorService(executorService);
+
+// Create the Pub/Sub access token cache.
+PubsubAccessTokenCache accessTokenCache = PubsubAccessTokenCache
+        .builder()
+        .setExecutorService(executorService)
+        .setCredentials("awesome-credentials")
+        .build();
+
+// Create a meter registry for metrics.
+MeterRegistry meterRegistry = new SimpleMeterRegistry();
+
+// Create the Reactor Netty HTTP client.
+HttpClient httpClient = HttpClient.create();
+
+// Create the Pub/Sub client.
+PubsubClient client = PubsubClient
+        .builder()
+        .setAccessTokenCache(accessTokenCache)
+        .setHttpClient(httpClient)
+        .setMeterRegistry(meterRegistry)
+        .build();
+```
+
+You can create a publisher and publish a message as follows:
+
+```java
+// Create the publisher.
+String projectName = "awesome-project";
+PubsubPublisherConfig publisherConfig = PubsubPublisherConfig
+        .builder()
+        .setProjectName(projectName)
+        .setTopicName("awesome-topic")
+        .build();
+PubsubPublisher publisher = PubsubPublisher
+        .builder()
+        .setClient(client)
+        .setConfig(publisherConfig)
+        .build();
+
+// Publish a message.
+publisher
+        .publish(new PubsubPublishRequest(
+                Collections.singletonList(
+                        new PubsubDraftedMessage(
+                                "Yolo like no is watching you!"
+                                        .getBytes(StandardCharsets.UTF_8)))))
+        .subscribeOn(scheduler)
+        .subscribe();
+```
+
+You can create a subscriber and start receiving messages from a subscription as
+follows:
+
+```java
+// Create the message consumer.
+PubsubSubscriber.PullResponseConsumer pullResponseConsumer =
+        pullResponse -> {
+            int messageCount = pullResponse.getReceivedAckableMessages().size();
+            System.out.format("Just got awesome %d messages!%n", messageCount);
+            List<String> ackIds = pullResponse
+                    .getReceivedAckableMessages()
+                    .stream()
+                    .map(PubsubReceivedAckableMessage::getAckId)
+                    .collect(Collectors.toList());
+            PubsubAckRequest ackRequest = new PubsubAckRequest(ackIds);
+            return Mono.just(ackRequest);
+        };
+
+// Create a subscriber.
+PubsubSubscriberConfig subscriberConfig = PubsubSubscriberConfig
+        .builder()
+        .setProjectName(projectName)
+        .setSubscriptionName("awesome-subscription")
+        .build();
+PubsubSubscriber subscriber = PubsubSubscriber
+        .builder()
+        .setClient(client)
+        .setConfig(subscriberConfig)
+        .setPullResponseConsumer(pullResponseConsumer)
```

Comment:
This API is not very "reactive", rather async. A reactive API would be to have a stream of `pullResponse`, so that the user can also do batching, windowing, backpressure operations, etc etc

Reply:
If I return a `Flux<PubsubPullResponse>` from `start()`, then the users will need to ACK themselves manually. Is that really a good idea? (Yep, sincerely asking.)

Reply:
You should know PubSub better than me, but I would say yes, that's normal, as long as the ACK API is easy to use.

You may have different "strategies" too, auto ack, manual ack, etc etc

Reply:
While working on this, I have noticed another issue: Once I give `ack()` responsibility to the user, I need to give the responsibility of the `rateLimit()` transformation (which needs to be chained to the end of the pull+ack stream) to them as well. :egg: But maybe this explicitness is something better.